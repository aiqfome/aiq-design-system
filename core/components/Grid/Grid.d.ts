import React from 'react';
import { SpaceProps, GridProps, LayoutProps, FontSizeProps, FontWeightProps } from 'styled-system';
export interface Props extends SpaceProps, LayoutProps, FontSizeProps, GridProps, FontWeightProps {
    color?: string;
}
export declare const GridStyled: import("styled-components").StyledComponent<"div", any, Props, never>;
export declare const Grid: React.FC<Props>;
//# sourceMappingURL=Grid.d.ts.map