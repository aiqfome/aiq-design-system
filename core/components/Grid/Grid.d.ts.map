{"version":3,"file":"Grid.d.ts","sourceRoot":"","sources":["../../../src/components/Grid/Grid.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAA;AAGzB,OAAO,EAGL,UAAU,EAEV,SAAS,EAET,WAAW,EAEX,aAAa,EAEb,eAAe,EAChB,MAAM,eAAe,CAAA;AAEtB,MAAM,WAAW,KACf,SAAQ,UAAU,EAChB,WAAW,EACX,aAAa,EACb,SAAS,EACT,eAAe;IACjB,KAAK,CAAC,EAAE,MAAM,CAAA;CACf;AAED,eAAO,MAAM,UAAU,uEAStB,CAAA;AAED,eAAO,MAAM,IAAI,EAAE,KAAK,CAAC,EAAE,CAAC,KAAK,CAEhC,CAAA"}