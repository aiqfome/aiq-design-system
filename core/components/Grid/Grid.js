"use strict";
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Grid = exports.GridStyled = void 0;
const react_1 = __importDefault(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const styled_components_1 = __importDefault(require("styled-components"));
const styled_system_1 = require("styled-system");
exports.GridStyled = styled_components_1.default.div `
  ${styled_system_1.color}
  ${styled_system_1.space}
  ${styled_system_1.layout}
  ${styled_system_1.fontSize}
  ${styled_system_1.fontWeight}
  ${styled_system_1.grid}

  display: grid;
`;
exports.Grid = (_a) => {
    var props = __rest(_a, []);
    return react_1.default.createElement(exports.GridStyled, Object.assign({}, props));
};
exports.GridStyled.propTypes = {
    color: prop_types_1.default.string
};
