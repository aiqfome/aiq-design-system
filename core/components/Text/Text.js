"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Text = void 0;
const prop_types_1 = __importDefault(require("prop-types"));
const styled_components_1 = __importStar(require("styled-components"));
const styled_system_1 = require("styled-system");
exports.Text = styled_components_1.default.span `
  ${styled_system_1.color}
  ${styled_system_1.space}
  ${styled_system_1.layout}
  ${styled_system_1.fontSize}
  ${styled_system_1.fontWeight}
  ${styled_system_1.typography}

  white-space: ${props => props.whiteSpace || 'normal'};

  ${({ cursor }) => cursor &&
    styled_components_1.css `
      cursor: ${cursor};
    `};

  ${({ truncate }) => truncate &&
    styled_components_1.css `
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
    `};
`;
exports.Text.propTypes = {
    cursor: prop_types_1.default.string,
    whiteSpace: prop_types_1.default.oneOf(['nowrap', 'normal', 'pre'])
};
