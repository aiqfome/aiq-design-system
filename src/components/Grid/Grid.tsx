import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import {
  color,
  space,
  SpaceProps,
  grid,
  GridProps,
  layout,
  LayoutProps,
  fontSize,
  FontSizeProps,
  fontWeight,
  FontWeightProps
} from 'styled-system'

export interface Props
  extends SpaceProps,
    LayoutProps,
    FontSizeProps,
    GridProps,
    FontWeightProps {
  color?: string
}

export const GridStyled = styled.div<Props>`
  ${color}
  ${space}
  ${layout}
  ${fontSize}
  ${fontWeight}
  ${grid}

  display: grid;
`

export const Grid: React.FC<Props> = ({ ...props }) => {
  return <GridStyled {...props} />
}

GridStyled.propTypes = {
  color: PropTypes.string
}
