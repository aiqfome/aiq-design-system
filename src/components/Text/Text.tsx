import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'
import {
  color,
  ColorProps,
  space,
  SpaceProps,
  layout,
  LayoutProps,
  fontSize,
  FontSizeProps,
  fontWeight,
  FontWeightProps,
  typography,
  TypographyProps
} from 'styled-system'

export interface Props
  extends ColorProps,
    SpaceProps,
    LayoutProps,
    FontSizeProps,
    FontWeightProps,
    TypographyProps {
  cursor?: string
  whiteSpace?: 'nowrap' | 'normal' | 'pre'
}

export const Text = styled.span<Props>`
  ${color}
  ${space}
  ${layout}
  ${fontSize}
  ${fontWeight}
  ${typography}

  ${({ cursor }) =>
    cursor &&
    css`
      cursor: ${cursor};
    `};
  white-space: ${props => props.whiteSpace || 'normal'};
`

Text.propTypes = {
  cursor: PropTypes.string,
  whiteSpace: PropTypes.oneOf(['nowrap', 'normal', 'pre'])
}
