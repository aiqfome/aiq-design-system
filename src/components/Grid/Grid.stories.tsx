import React from 'react'

import { Grid } from './Grid'
import { Text } from '../Text'

export default {
  component: Grid,
  title: 'Grid'
}

export const basic: React.FC = () => (
  <Grid color='primary'>
    <Text>Design System</Text>
    <Text>Design System</Text>
    <Text>Design System</Text>
    <Text>Design System</Text>
  </Grid>
)

export const templateColumns: React.FC = () => (
  <Grid color='primary' gridTemplateColumns='1fr 2fr'>
    <Text>Design System</Text>
    <Text>Design System</Text>
    <Text>Design System</Text>
    <Text>Design System</Text>
  </Grid>
)
