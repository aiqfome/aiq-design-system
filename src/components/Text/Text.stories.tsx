import React from 'react'
import { withKnobs, text, number } from '@storybook/addon-knobs'

import { Text } from './Text'
import { Flex } from '../Flex'

export default {
  component: Text,
  title: 'Text',
  decorators: [withKnobs]
}

export const basic: React.FC = () => (
  <Text
    color={text('color', 'primary')}
    fontSize={number('fontSize', 16)}
    cursor={text('cursor', '')}
    fontWeight={text('fontWeight', 'medium')}
  >
    {text('text', 'Aiqfome')}
  </Text>
)

export const cursor: React.FC = () => (
  <Text cursor='pointer'>Design System</Text>
)

export const WhiteSpace: React.FC = () => (
  <Flex width='50px' border='1px solid black'>
    <Text whiteSpace='nowrap'>Design System</Text>
  </Flex>
)
