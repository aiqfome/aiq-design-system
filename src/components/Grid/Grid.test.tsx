import React from 'react'

import { Grid } from './Grid'
import { Text } from '../Text'

import { render } from '../utils/test/render'

describe('must match with the previous snapshot', () => {
  test('snapshot renders', () => {
    const component = render(
      <Grid>
        <Text>Grid component!</Text>
      </Grid>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
