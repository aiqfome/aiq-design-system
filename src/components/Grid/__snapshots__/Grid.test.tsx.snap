// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must match with the previous snapshot snapshot renders 1`] = `
<div
  className="sc-AxjAm cFeSPz"
>
  <span
    className="sc-AxirZ bGDsjS"
  >
    Grid component!
  </span>
</div>
`;
